Lombiq.Tests.UI/Services/UITestExecutionSession.cs

```diff
@@ -140,6 +131,33 @@ public async Task<bool> ExecuteAsync(int retryCount, string dumpRootPath)
             return false;
         }
 
+        private Exception ProcessException(Exception exception)
+        {
+            if (exception is PageChangeAssertionException pageChangeAssertionException)
+            {
+                _testOutputHelper.WriteLineTimestampedAndDebug(pageChangeAssertionException.Message);
+                exception = pageChangeAssertionException.InnerException;
+
+                if (exception is HtmlValidationAssertionException htmlValidationAssertionException &&
+                    htmlValidationAssertionException.HtmlValidationConfiguration.CreateReportOnFailure)
+                {
+                    _testOutputHelper.WriteLineTimestampedAndDebug(
+                        "Check the HTML validation report in the failure dump for details.");
+                }
+            }
+            else
+            {
+                _testOutputHelper.WriteLineTimestampedAndDebug(
+                    $"An exception has occurred while interacting with the page {_context.GetPageTitleAndAddress()}.");
+            }
+
+            _testOutputHelper.WriteLineTimestampedAndDebug($"The test failed with the following exception: {exception}");
+
+            if (exception is SetupFailedFastException) throw exception;
```

Comment:
`throw exception` will reset the stack trace: https://stackoverflow.com/a/730255/220230

Reply:
wtf!! That makes no sense. Anyway, I've reverted the auxiliary method.